### PR TITLE
[RFC] lint: Removing dead initializations

### PR DESCRIPTION
**Pull request by arcuru**
This is just a small refactor to eliminate the 2 clang analysis warnings.

### CODE REVIEW

**Comment by phanimahesh** [comment]
Thanks! Can you please amend your commit message to include the issue being fixed and also include `[RFC]` in the title so that contributors can filter out PRs that are ready.

**Comment by jamessan** [comment]
`[RFC]` should only be in the PR title, not the commit itself.

**Comment by phanimahesh** [comment]
Yes, I should have been more explicit, I meant the PR title and the commit message.

File: src/nvim/api/private/helpers.c [hunk 1]
```diff
@@ -368,11 +368,10 @@ static inline void typval_encode_list_start(EncodedData *const edata,
                                             const size_t len)
   FUNC_ATTR_ALWAYS_INLINE FUNC_ATTR_NONNULL_ALL
 {
-  const Object obj = OBJECT_INIT;
   kv_push(edata->stack, ARRAY_OBJ(((Array) {
     .capacity = len,
     .size = 0,
-    .items = xmalloc(len * sizeof(*obj.data.array.items)),
+    .items = xmalloc(len * sizeof(*((Object)OBJECT_INIT).data.array.items)),
```
Comment (justinmk) [inline src/nvim/api/private/helpers.c]: How about `((Object *)NULL)->data.array.items` ? http://stackoverflow.com/a/3864612/152142

Comment (patricksjackson) [inline src/nvim/api/private/helpers.c]: That would be better, I wasn't aware you could even do that. I'll update with that change.

**Comment by justinmk** [comment]
Merged, thanks

**Pull request closed without merging**